src/server/main_service.cc

```diff
@@ -665,7 +665,7 @@ void Service::Init(util::AcceptServer* acceptor, std::vector<facade::Listener*>
 
   shard_set->Init(shard_num, !opts.disable_time_update);
   const auto tcp_disabled = GetFlag(FLAGS_port) == 0u;
-  if (!tcp_disabled) {
+  if (!tcp_disabled && !listeners.empty()) {
```

Comment:
please add a comment that we assume here that the main interface comes first. Also, please CHECK that it's not an admin interface.

Reply:
I did add a comment, in `dfly_main` but I will add one here as well